kube-runtime/src/scheduler.rs

```diff
@@ -392,4 +400,56 @@ mod tests {
         scheduler.next().now_or_never().unwrap().unwrap();
         assert!(poll!(scheduler.next()).is_pending());
     }
+
+    #[tokio::test]
+    async fn scheduler_should_overwrite_message_with_soonest_version() {
+        // Message type that is always considered equal to itself
+        #[derive(Derivative, Eq, Clone, Debug)]
+        #[derivative(PartialEq, Hash)]
+        struct SingletonMessage(#[derivative(PartialEq = "ignore", Hash = "ignore")] u8);
```

Comment:
lol what  a place to put a proc macro.

Reply:
:p

Wouldn't pull in the dependency just for this, but since we already have it...